server.go

```diff
@@ -86,19 +88,41 @@ type server struct {
 	errorChan   chan struct{}
 	closed      bool
 
-	sessionQueue    chan Session
+	sessionQueue    chan quicSession
 	sessionQueueLen int32 // to be used as an atomic
 
 	logger utils.Logger
 }
 
-var _ Listener = &server{}
-var _ unknownPacketHandler = &server{}
+var _ Listener = &baseServer{}
+var _ unknownPacketHandler = &baseServer{}
+
+type earlyServer struct{ *baseServer }
```

Comment:
Maybe my Go is rusty, but can't this just be an alias?

Reply:
I think I need a struct here to define the `Accept()` method, don't I?